main.py

```diff
@@ -83,34 +92,171 @@ def run(driver, shoe_type, username, password, url, shoe_size, login_time=None,
                 LOGGER.exception("Failed to select shoe size: " + str(e))
                 continue
-
+                
             try:
                 click_buy_button(driver=driver)
             except Exception as e:
-                LOGGER.exception("Failed to click buy button: " + str(e))
+                LOGGER.exception("Failed to click buy button: " + str(e))                                
                 six.reraise(Exception, e, sys.exc_info()[2])
-
-            if select_payment:
+                
+            #Loop to determine which element appears first
+            #Limit this loop as "Verify Phone Number" dialog may appear
+            checkout_num_retries_attempted = 0
+            checkout_num_retries = 25
+            while True:
+                try:
+                    check_add_new_address_button(driver=driver)
+                    LOGGER.info("New Address button appeared!")
+                    break
+                except Exception as e:
+                    LOGGER.exception("Failed visibility check for Add New Address button: " + str(e))
+                    
                 try:
-                    select_payment_option(driver=driver)
+                    check_shipping(driver=driver)
+                    LOGGER.info("Shipping options appeared!")
+                    skip_add_address = True
+                    break
                 except Exception as e:
-                    LOGGER.exception("Failed to select payment option: " + str(e))
+                    LOGGER.exception("Failed visibility check for Shipping options: " + str(e))
+                    
+                try:
+                    check_payment(driver=driver)
+                    LOGGER.info("Payment appeared!")
+                    skip_add_address = True
+                    skip_select_shipping = True
+                    break
+                except Exception as e:
+                    LOGGER.exception("Failed visibility check for Payment: " + str(e))
+                
+                try:
+                    xpath = SUBMIT_BUTTON_XPATH
+                    check_submit_button(driver=driver, xpath_o=xpath)
+                    LOGGER.info("Submit Button appeared!")
+                    skip_add_address = True
+                    skip_select_shipping = True
+                    skip_payment = True
+                    break
+                except Exception as e:
+                    LOGGER.exception("Failed visibility check for Submit Button: " + str(e))
+                    
+                if checkout_num_retries_attempted < checkout_num_retries:
+                    checkout_num_retries_attempted += 1
+                    continue
+                else:
+                    LOGGER.info("Too many iterations through checkout element check. Terminating check...")
+                    break
+                    
+            if skip_add_address is False and shipping_address:
```

Comment:
This function is getting unwieldy and hard to read. Can you break some of this logic up into smaller functions?